ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/events/AbortTxnHandler.java

```diff
@@ -39,6 +48,19 @@ public void handle(Context withinContext) throws Exception {
     if (!ReplUtils.includeAcidTableInDump(withinContext.hiveConf)) {
       return;
     }
+
+     if (ReplUtils.filterTransactionOperations(withinContext.hiveConf)) {
+       String contextDbName = StringUtils.normalizeIdentifier(withinContext.replScope.getDbName());
+       GetTxnWriteIdsRequest request = new GetTxnWriteIdsRequest(eventMessage.getTxnId());
+       request.setDbName(contextDbName);
+       GetTxnWriteIdsResponse response = withinContext.db.getMSC().getTxnWriteIds(request);
```

Comment:
Could we do this without calling the metastore?
All of requested writeIds should be available during the compilation phase

Reply:
To move this during compilation, there are 2 difficulties:
1. Would need to add a new field/status to AbortTxnEvent, this field would indicate whether or not this aborted txn involved writeids. REPL DUMP could filter based on this setting.
2. The AbortTxnEvent is logged via TxnHandler.abort_txn(), so AbortTxnRequest would need to be modified to pass the writeid. AbortTxnRequest is a Thrift object. OR, abort_txn() could do the same HMS lookup that is currently done in AbortTxnHandler().

Since this metastore call is done during REPL DUMP, would it be ok to live with this inefficiency and fix later with a more optimal implementation?

Reply:
Running this from the client side the changes might not be atomic, which could cause issues later.
I think all of these calculations should be done on the HMS side when storing the event

Reply:
Ok, I changed how this is done.
TxnHandler.abortTxn() will check if the txn had allocated writeids, and attach the databases to the AbortTxnEvent. During REPL DUMP, when processing the AbortTxnEvent, add the event to the dump only if the AbortTxnEvent affected the database being dumped.

Reply:
Can we remove the HMS API change now?